### DISCUSSION TITLE
How to redirect print_color as plain text

### DISCUSSION THREAD

**Discussion by eugenesvk** [xonsh/xonsh#5614] [Q&A]
I use `print_color` in aliases and when running `myalias > tofile.txt` `print_color` just prints to the terminal instead of being redirected.
I think `print_color(f"{{GREEN}}print_color{{RESET}} to file.stdout", file=sys.stdout)` is the way to allow it to be redirected, but then the output is ugly `[0m[?7h[0;38;2;0;176;0mprint_color[0m to file.stdout` 

Can I detect inside a script that it's being redirected and use a different function that prints noncolored output to file?

**Comment by anki-code** [comment]
I think it's new feature to detect that output is redirect to file in the alias body i.e.:
```
def alias(args, input, output)
    if output is file/redirect
        print
    else
        print_color
```

Nice feature. PR is welcome!

PS. Command in xonsh is [SubprocSpec](https://github.com/xonsh/xonsh/blob/32a085543a7594a39fa61062cbaa06cc540f23fc/xonsh/procs/specs.py#L319). For what is happening in case of redirect you can see [resolve_redirect](https://github.com/xonsh/xonsh/blob/32a085543a7594a39fa61062cbaa06cc540f23fc/xonsh/procs/specs.py#L647).
